### DISCUSSION TITLE
Help with SSR support...

### DISCUSSION THREAD

**Discussion by bvkimball** [buildo/bento-design-system#543] [Q&A] [answered]
I am using the components with Remix, i tried to wrap the app with `@react-aria/ssr` but i still get the following error.

> When server rendering, you must wrap your application in an <SSRProvider> to ensure consistent ids are generated between the client and server.

I am using the latest `@react-aria/ssr` and the latest bento.

**Comment by gabro** [comment, marked as answer]
Hi @bvkimball, thanks for reporting! Would you be able to provide a small repro (a fresh repo with the setup would be ideal)?
Otherwise I'll try to repro myself in the next days.
I'm actually tinkering with remix too, so it would be good to have things working properly with SSR :)

**Reply by bvkimball** [reply]
https://codesandbox.io/p/sandbox/cocky-leftpad-diwlbk?file=%2Fapp%2Fentry.server.tsx&selection=%5B%7B%22endColumn%22%3A28%2C%22endLineNumber%22%3A34%2C%22startColumn%22%3A28%2C%22startLineNumber%22%3A34%7D%5D

Well this seems to work so.... must be something i am doing 😮‍💨 

If the codesandbox says "Upgrade Required" that just means you need to switch the 8002 in the url to 3000.

**Reply by gabro** [reply]
I see. Please let me know if you find the root cause of your issue. I'm totally open to changes to Bento to make this less error prone

**Reply by gabro** [reply]
@bvkimball in the meantime I've opened https://github.com/buildo/bento-design-system/pull/548 to make SSR support a bit more ergonomic

**Reply by bvkimball** [reply]
@gabro As i have dug deeper, bento also needs a check to see if it is SSR so that it knows to skip any reference to `document`, which can be found anywhere a portal is created. Will add comment to #584 with details.
